src-tauri/src/main.rs

```diff
@@ -1,8 +1,35 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+use std::path::Path;
+
+use tauri::api::path::home_dir;
+
 fn main() {
   tauri::Builder::default()
+    .invoke_handler(tauri::generate_handler![
+      list
+      ])
     .run(tauri::generate_context!())
     .expect("error while running tauri application");
 }
+
+/// list all possible README.md locations in dirs of the save dir
+#[tauri::command]
+async fn list() -> Result<Vec<String>, String> {
+  let base_path = Path::new(home_dir().unwrap().as_path()).join(".notes");
+  
+  let mut notes = Vec::<String>::new();
+
+  // read all files in dir
+  let dir_entries = std::fs::read_dir(base_path).map_err(|err| err.to_string())?;
+  for dir_entry in dir_entries {
+    let dir_entry = dir_entry.map_err(|err| err.to_string())?;
+    let file_type = dir_entry.file_type().map_err(|err| err.to_string())?;
+    if file_type.is_dir() {
```

Comment:
Please check if the directory contains a file named `README.md` and add only directories that contain such a file. Please refer to [note.js's implementation](https://github.com/falk-werner/note.js/blob/main/src/backend/note.js).